runtime/parser/parse_metrics_view.go

```diff
@@ -820,14 +823,28 @@ func (p *Parser) parseMetricsView(node *Node) error {
 			continue
 		}
 
+		// all dict fields should be defined or none
+		if dim.LookupTable == "" && dim.LookKeyColumn == "" && dim.LookValueColumn == "" { // nolint:revive // avoids checking all permutations
+			// do nothing
+		} else if dim.LookupTable != "" && dim.LookKeyColumn != "" && dim.LookValueColumn != "" {
+			if dim.Column == "" {
+				return fmt.Errorf("column is required if lookup fields are defined, containing values to be looked up")
+			}
```

Comment:
Could we try to support `expression`? I may be wrong, but I think the UI sometimes defaults to `expression:` instead of `column:` now

Reply:
https://github.com/rilldata/rill/pull/7051#discussion_r2053346496